service/src/main/java/eu/europa/ec/fisheries/uvms/activity/fa/entities/FishingTripEntity.java

```diff
@@ -38,17 +38,18 @@ This file is part of the Integrated Fisheries Data Management (IFDM) Suite. The
 import java.util.Set;
 
 @NamedQueries({
-        @NamedQuery(name = FishingTripEntity.FIND_CURRENT_TRIP,
+        @NamedQuery(name = FishingTripEntity.FIND_TRIPS_FOR_VESSEL_ORDERED_BY_DATE_LATEST_FIRST,
                 query = "SELECT fa.fishingTrip FROM FishingActivityEntity fa " +
                         "INNER JOIN fa.faReportDocument frd " +
                         "INNER JOIN frd.vesselTransportMeans vtm " +
                         "INNER JOIN vtm.vesselIdentifiers vi " +
                         "WHERE vi.vesselIdentifierId = :vesselId " +
-                        "AND vi.vesselIdentifierSchemeId = :vesselSchemeId")
+                        "AND vi.vesselIdentifierSchemeId = :vesselSchemeId " +
+                        "ORDER BY frd.acceptedDatetime DESC")
 
 		/*
 		//TODO: We need to rewrite this...
-		@NamedQuery(name = FishingTripIdentifierEntity.FIND_PREVIOUS_TRIP,
+		@NamedQuery(name = FishingTripIdentifierEntity.FIND_TRIPS_BEFORE_TRIP_WITH_ID,
```

Comment:
max(sfrd.acceptedDatetime)  =>  min(sfrd.acceptedDatetime)    or?   If trips cannot overlap it doesn't matter, but to be on the safe side
And remove TODO, if you find it ok

Reply:
I'm leaving this here for now since we don't have a good way to test it right now.